core/src/nodes/node.rs

```diff
@@ -280,28 +280,34 @@ where
 
         // Polling outbound substreams.
         // We remove each element from `outbound_substreams` one by one and add them back.
-        for n in (0..self.outbound_substreams.len()).rev() {
-            let (user_data, mut outbound) = self.outbound_substreams.swap_remove(n);
-            match self.muxer.poll_outbound(&mut outbound) {
-                Ok(Async::Ready(Some(substream))) => {
-                    let substream = muxing::substream_from_ref(self.muxer.clone(), substream);
-                    self.muxer.destroy_outbound(outbound);
-                    return Ok(Async::Ready(Some(NodeEvent::OutboundSubstream {
-                        user_data,
-                        substream,
-                    })));
-                }
-                Ok(Async::Ready(None)) => {
-                    self.outbound_state = StreamState::Closed;
-                    self.muxer.destroy_outbound(outbound);
-                    return Ok(Async::Ready(Some(NodeEvent::OutboundClosed { user_data })));
-                }
-                Ok(Async::NotReady) => {
-                    self.outbound_substreams.push((user_data, outbound));
-                }
-                Err(err) => {
-                    self.muxer.destroy_outbound(outbound);
-                    return Err(err);
+        if self.outbound_state == StreamState::Open {
+            if self.outbound_substreams.is_empty() {
+                self.outbound_state = StreamState::Closed;
+                return Ok(Async::Ready(None))
```

Comment:
This is clearly not correct. We don't want to close the NodeStream right off the bat here, but what is the correct return value here? `NotReady`?